test/built-ins/RegExp/CharacterClassEscapes/character-class-digit-class-escape-flags-u.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018 Leo Balter.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: prod-CharacterClassEscape
+description: >
+    Compare range for Digit class escape, \\d with flags u
+info: |
+    This is a generated test, please checkout https://github.com/bocoup/test262-regexp-generator
+    for any changes.
+
+    CharacterClassEscape[U] ::
+        d
+        D
+        s
+        S
+        w
+        W
+
+    21.2.2.12 CharacterClassEscape
+    
+    The production CharacterClassEscape :: d evaluates as follows:
+        Return the ten-element set of characters containing the characters 0 through 9 inclusive.
+    The production CharacterClassEscape :: D evaluates as follows:
+        Return the set of all characters not included in the set returned by CharacterClassEscape :: d.
+    The production CharacterClassEscape :: s evaluates as follows:
+        Return the set of characters containing the characters that are on the right-hand side of
+        the WhiteSpace or LineTerminator productions.
+    The production CharacterClassEscape :: S evaluates as follows:
+        Return the set of all characters not included in the set returned by CharacterClassEscape :: s.
+    The production CharacterClassEscape :: w evaluates as follows:
+        Return the set of all characters returned by WordCharacters().
+    The production CharacterClassEscape :: W evaluates as follows:
+        Return the set of all characters not included in the set returned by CharacterClassEscape :: w.
+features: [String.fromCodePoint]
+---*/
+
+var re = /\d/u;
+var matchingRange = /[0-9]/u;
+var msg = '"\\u{REPLACE}" should be in range for \\d with flags u';
+
+var i;
+var fromEscape, fromRange, str;
+for (i = 0; i < 0x10FFFF; i++) {
+
+    str = String.fromCodePoint(i);
+    fromEscape = !str.replace(re, 'test262');
+    fromRange = !str.replace(re, 'test262');
+    assert.sameValue(fromEscape, fromRange, msg.replace('REPLACE', i));
```

Comment:
You want `i.toString(16)`

Reply:
nit: maybe use `codePoint` or `cp` instead of `i` for all these tests.

`i` is especially confusing on this line, where it’s close to `.replace` which puts my mind into RegExp mode. “Why does this need the `i` flag?”